app/assets/javascripts/govuk_publishing_components/lib/cookie-settings.js

```diff
@@ -12,31 +12,39 @@ window.GOVUK.Modules = window.GOVUK.Modules || {};
     document.querySelector('form[data-module=cookie-settings]')
       .addEventListener('submit', this.$module.submitSettingsForm)
 
-    this.setInitialFormValues()
+    if (window.GOVUK.useSingleConsentApi) {
+      window.GOVUK.singleConsent.init(this.setInitialFormValues.bind(this))
+    } else {
+      this.setInitialFormValues()
+    }
   }
 
   CookieSettings.prototype.setInitialFormValues = function () {
     if (!window.GOVUK.cookie('cookies_policy')) {
-      window.GOVUK.setDefaultConsentCookie()
+      if (!window.GOVUK.useSingleConsentApi) {
+        window.GOVUK.setDefaultConsentCookie()
+      }
     }
 
     var currentConsentCookie = window.GOVUK.cookie('cookies_policy')
-    var currentConsentCookieJSON = JSON.parse(currentConsentCookie)
+    if (currentConsentCookie) {
+      var currentConsentCookieJSON = JSON.parse(currentConsentCookie)
 
-    // We don't need the essential value as this cannot be changed by the user
-    delete currentConsentCookieJSON.essential
+      // We don't need the essential value as this cannot be changed by the user
+      delete currentConsentCookieJSON.essential
```

Comment:
Forgot to mention this. Here you're deleting the `essential` cookie but not reintroducing it into the cookie object when the user submits the form.

Saving cookies isn't working properly in IE11 on this branch - it saves sucessfully, but then when I go to the homepage, the cookie settings I have in `document.cookie` are different to the ones that I saved.

Could this be because `essential` is missing from the cookie when the form is saved, causing something odd?

Reply:
Is this a problem with this branch, or more generally? I don't think I introduced this change. Would it be alright to log this as an issue and for it to not block this change?

Reply:
No worries - I've just tried it again and I think it's actually a weird IE11 caching thing. If I update my cookies, and go back to the homepage, the homepage gives me the old cookie state unless I add a cachebust query string to the URL. So don't worry about it